nestjs/src/courses/team-distribution/team-distribution.controller.ts

```diff
@@ -100,4 +123,56 @@ export class TeamDistributionController {
       await this.studentsService.deleteStudentFromTeamDistribution(studentId, teamDistribution);
     }
   }
+
+  @Get('/:id/detailed')
+  @UseGuards(RoleGuard)
+  @ApiOkResponse({ type: TeamDistributionDetailedDto })
+  @ApiOperation({ operationId: 'getCourseTeamDistributionDetailed' })
+  @RequiredRoles([CourseRole.Student, CourseRole.Manager, Role.Admin])
+  public async getCourseTeamDistributionDetailed(
+    @Req() req: CurrentRequest,
+    @Param('courseId', ParseIntPipe) courseId: number,
+    @Param('id', ParseIntPipe) id: number,
+  ) {
+    const studentId = req.user.courses[courseId]?.studentId;
+    const isManager = req.user.isAdmin || req.user.courses[courseId]?.roles.includes(CourseRole.Manager);
+    let team;
+    if (studentId) {
+      const student = await this.studentsService.getStudentDetailed(studentId);
+      if (
+        !student.teamDistribution.find(d => d.id === id) &&
+        !student.teams.find(t => t.teamDistributionId === id) &&
+        !isManager
+      ) {
+        throw new ForbiddenException();
+      }
+      const data = student.teams.find(t => t.teamDistributionId === id);
+      if (data) {
+        team = await this.teamService.findByIdDetailed(data.id);
+        team = new TeamDto(team);
+      }
+    }
+    const distribution = await this.teamDistributionService.getDistributionDetailedById(id);
+    return new TeamDistributionDetailedDto(distribution, team);
+  }
+
+  @Get('/:id/students')
+  @UseGuards(RoleGuard)
+  @ApiOkResponse({ type: [TeamDistributionStudentDto] })
+  @ApiOperation({ operationId: 'getStudentsWithoutTeam' })
+  @RequiredRoles([CourseRole.Student, CourseRole.Manager, Role.Admin])
```

Comment:
just to make sure you want to give this access to all students

Reply:
fixed